src/file.ts

```diff
@@ -2097,7 +2098,18 @@ class File extends ServiceObject<File> {
 
     const fileStream = this.createReadStream(options);
     let receivedData = false;
-    if (destination) {
+
+    // Skip directory objects as they cannot be written to local filesystem
+    /**
+     * Temporary comment:
+     * Could consider returning an error, but TransferManager will have to handle rejected promises in Promises.all.
+     * Example: Promise.allSettled(promises) or Promise.all(promises.map(p => p.catch(e => e)))
+     */
+    if (destination && destination.endsWith('/')) {
+      callback?.(null, Buffer.alloc(0));
+    } else if (destination) {
```

Comment:
With how GCS works, empty directory objects get returned to `getFiles`. This causes issues as TransferManager will attempt to write both `folder1/` and `folder1/foo.txt`. 

Writing `folder1/` throws this:
```
Error: EISDIR: illegal operation on a directory, open '/folder1/'
```

In this if statement, we skip writing directory objects, and just return an empty buffer of size zero. Could also choose to return an Error, however, that means TransferManager has to handle rejected promises.